### PR TITLE
e2e test for kfctl istio

### LINKED ISSUE #2891: kfctl e2e test for istio

### LINKED ISSUE #3049: kfctl install istio fails with GOOGLE_APPLICATION_CREDENTIALS
Just like what we do in [test](https://github.com/kubeflow/kubeflow/blob/master/testing/kfctl/kfctl_go_test.py): 
1. use GOOGLE_APPLICATIOB_CREDENTIALS to point to a service account
1. gcloud activate that
1. set email to that service account

Installing istio failed to create clusterrole.
However, there is already a binding for service account and cluster admin.

Maybe it's because when installing istio we get another identity / wrong way getting credential?

### PR DESCRIPTION
**Pull request by lluunn**
fork from #2984

Fix #3049 
Fix #2891 

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3054)

### CODE REVIEW

**Comment by lluunn** [comment]
test passed!
/assign @gabrielwen @kunmingg

File: testing/kfctl/conftest.py [hunk 1]
```diff
@@ -21,6 +21,10 @@ def pytest_addoption(parser):
       "--use_basic_auth", action="store", default="False",
       help="Use basic auth.")
 
+  parser.addoption(
+      "--use_istio", action="store", default="False",
```
Comment (gabrielwen) [inline testing/kfctl/conftest.py]: shouldn't you use something like this?
https://github.com/kubeflow/testing/blob/master/py/kubeflow/testing/get_kf_testing_cluster.py#L229-L237

File: bootstrap/pkg/kfapp/gcp/gcp.go [hunk 2]
```diff
@@ -664,23 +664,29 @@ func (gcp *Gcp) updateDM(resources kftypes.ResourceEnum) error {
...
-		return &kfapis.KfError{
-			Code: err.(*kfapis.KfError).Code,
-			Message: fmt.Sprintf("Build ClientConfig error: %v",
-				err.(*kfapis.KfError).Message),
+		if _, err := os.Stat(kftypes.KubeConfigPath()); !os.IsNotExist(err) {
+			gcp.AddNamedContext()
 		}
 	}
+	// Get client for kube config.
+	client := kftypes.GetConfig()
```
Comment (gabrielwen) [inline bootstrap/pkg/kfapp/gcp/gcp.go]: could you see if this method can be used?
https://github.com/kubeflow/kubeflow/blob/de1fc808b40f58e4bbb7199fedb68df29d412c98/bootstrap/pkg/utils/gcputils.go#L80

ideally we should isolate between GCP and k8s.

File: bootstrap/pkg/kfapp/gcp/gcp.go [hunk 3]
```diff
@@ -664,23 +664,29 @@ func (gcp *Gcp) updateDM(resources kftypes.ResourceEnum) error {
...
-		gcp.Spec.Zone, gcp.Name, gcp.tokenSource)
-	if err != nil {
-		return &kfapis.KfError{
-			Code: err.(*kfapis.KfError).Code,
-			Message: fmt.Sprintf("Configure K8s is failed: %v",
-				err.(*kfapis.KfError).Message),
+	// kfctl only
+	// Setup kube config
+	if gcp.isCLI {
+		// TODO(#2604): Need to create a named context.
```
Comment (gabrielwen) [inline bootstrap/pkg/kfapp/gcp/gcp.go]: could you help remove this line?

File: bootstrap/pkg/kfapp/gcp/gcp.go (hunk 2 above)
Comment (lluunn) [reply bootstrap/pkg/kfapp/gcp/gcp.go]: That creates clientcmdapi.Config, but we need rest.Config

File: testing/kfctl/conftest.py (hunk 1 above)
Comment (lluunn) [reply testing/kfctl/conftest.py]: it's a different module called pytest. See above for basic auth

File: bootstrap/pkg/kfapp/gcp/gcp.go (hunk 2 above)
Comment (gabrielwen) [reply bootstrap/pkg/kfapp/gcp/gcp.go]: then it's this one:
https://github.com/kubeflow/kubeflow/blob/de1fc808b40f58e4bbb7199fedb68df29d412c98/bootstrap/pkg/utils/gcputils.go#L60

**Comment by lluunn** [comment]
failure on kfctl.sh test

/retest

**Pull request merged**